src/H5Tconv_complex.c

```diff
@@ -0,0 +1,2323 @@
+                else {
+                    assert(realval_type == H5T_CONV_FLOAT_SPECVAL_NAN ||
+                           imagval_type == H5T_CONV_FLOAT_SPECVAL_NAN);
+                    except_type = H5T_CONV_EXCEPT_NAN;
+                }
+
+                except_ret = (conv_ctx->u.conv.cb_struct.func)(except_type, conv_ctx->u.conv.src_type_id,
+                                                               conv_ctx->u.conv.dst_type_id, src_rev, d,
+                                                               conv_ctx->u.conv.cb_struct.user_data);
+            }
+
+            if (except_ret == H5T_CONV_UNHANDLED) {
+                if (realval_type == H5T_CONV_FLOAT_SPECVAL_POSINF ||
+                    realval_type == H5T_CONV_FLOAT_SPECVAL_NEGINF ||
+                    imagval_type == H5T_CONV_FLOAT_SPECVAL_POSINF ||
+                    imagval_type == H5T_CONV_FLOAT_SPECVAL_NEGINF) {
+                    H5T__bit_copy(d, dst_atomic.u.f.sign, s, src_atomic.u.f.sign, (size_t)1);
+                    H5T__bit_copy(d + dst_part_size, dst_atomic.u.f.sign, s + src_part_size,
+                                  src_atomic.u.f.sign, (size_t)1);
+
+                    if (realval_type == H5T_CONV_FLOAT_SPECVAL_POSINF ||
+                        realval_type == H5T_CONV_FLOAT_SPECVAL_NEGINF) {
+                        H5T__bit_set(d, dst_atomic.u.f.epos, dst_atomic.u.f.esize, true);
+                        H5T__bit_set(d, dst_atomic.u.f.mpos, dst_atomic.u.f.msize, false);
+                        /* If the destination has no implied mantissa bit, we'll need to set
+                         * the 1st bit of mantissa to 1. The Intel-Linux "long double" is
+                         * this case. */
+                        if (H5T_NORM_NONE == dst_atomic.u.f.norm)
+                            H5T__bit_set(d, dst_atomic.u.f.mpos + dst_atomic.u.f.msize - 1, (size_t)1, true);
+                    }
+                    if (imagval_type == H5T_CONV_FLOAT_SPECVAL_POSINF ||
+                        imagval_type == H5T_CONV_FLOAT_SPECVAL_NEGINF) {
+                        H5T__bit_set(d + dst_part_size, dst_atomic.u.f.epos, dst_atomic.u.f.esize, true);
+                        H5T__bit_set(d + dst_part_size, dst_atomic.u.f.mpos, dst_atomic.u.f.msize, false);
+                        /* If the destination has no implied mantissa bit, we'll need to set
+                         * the 1st bit of mantissa to 1. The Intel-Linux "long double" is
+                         * this case. */
+                        if (H5T_NORM_NONE == dst_atomic.u.f.norm)
+                            H5T__bit_set(d + dst_part_size, dst_atomic.u.f.mpos + dst_atomic.u.f.msize - 1,
+                                         (size_t)1, true);
+                    }
+                }
+                else {
+                    /* There are many NaN values, so we just set all bits of the significand. */
+                    if (realval_type == H5T_CONV_FLOAT_SPECVAL_NAN) {
```

Comment:
Would probably be worthwhile setting up globals for NANs also (in additional to infinities), so that memcpy() could be used here to speed things up.